x/dex/module.go

```diff
@@ -270,59 +258,69 @@ func (am AppModule) endBlockForContract(ctx sdk.Context, contractAddr string) {
 	span.SetAttributes(attribute.String("contract", contractAddr))
 	defer span.End()
 
+	typedContractAddr := types.ContractAddress(contractAddr)
 	registeredPairs := am.keeper.GetAllRegisteredPairs(ctx, contractAddr)
 	_, currentEpoch := am.keeper.IsNewEpoch(ctx)
 
 	am.keeper.HandleEBLiquidation(spanCtx, ctx, am.tracingInfo.Tracer, contractAddr, registeredPairs)
 	am.keeper.HandleEBCancelOrders(spanCtx, ctx, am.tracingInfo.Tracer, contractAddr, registeredPairs)
 	am.keeper.HandleEBPlaceOrders(spanCtx, ctx, am.tracingInfo.Tracer, contractAddr, registeredPairs)
 
-	am.keeper.OrderCancellations[contractAddr] = map[string]*dexcache.OrderCancellations{}
 	for _, pair := range registeredPairs {
-		am.keeper.OrderCancellations[contractAddr][pair.String()] = dexcache.NewOrderCancellations()
-		orders := am.keeper.Orders[contractAddr][pair.String()]
-		ctx.Logger().Info(pair.String())
-		ctx.Logger().Info(fmt.Sprintf("Orders %s", am.keeper.Orders))
-		ctx.Logger().Info(fmt.Sprintf("Number of LB: %d, LS: %d, MB: %d, MS: %d", len(orders.LimitBuys), len(orders.LimitSells), len(orders.MarketBuys), len(orders.MarketSells)))
-		allExistingBuys := am.keeper.GetAllLongBookForPair(ctx, contractAddr, pair.PriceDenom, pair.AssetDenom)
-		allExistingSells := am.keeper.GetAllShortBookForPair(ctx, contractAddr, pair.PriceDenom, pair.AssetDenom)
+		typedPairStr := types.GetPairString(&pair)
+		orders := am.keeper.MemState.GetBlockOrders(typedContractAddr, typedPairStr)
```

Comment:
random q, theoretically this memstate is still redundant as the keeper also doesn't store directly to disk right? what's the reason we're keeping the MemState layer rn?

Reply:
we need memstate just to retain information from `DeliverTx` to `EndBlock`. It can be discarded after `EndBlock` runs